gcp-resources/src/test/java/io/opentelemetry/contrib/gcp/resource/GCPResourceProviderTest.java

```diff
@@ -0,0 +1,402 @@
+        EnumSet.of(
+            GCPPlatformDetector.SupportedPlatform.GOOGLE_CLOUD_RUN,
+            GCPPlatformDetector.SupportedPlatform.GOOGLE_CLOUD_FUNCTIONS);
+    if (!serverlessPlatforms.contains(platform)) {
+      throw new IllegalArgumentException();
+    }
+    Map<String, String> mockAttributes =
+        new HashMap<>(
+            ImmutableMap.of(
+                SERVERLESS_COMPUTE_NAME, "serverless-app",
+                SERVERLESS_COMPUTE_REVISION, "v2",
+                SERVERLESS_COMPUTE_INSTANCE_ID, "serverless-instance-id",
+                SERVERLESS_COMPUTE_CLOUD_REGION, "us-central1",
+                SERVERLESS_COMPUTE_AVAILABILITY_ZONE, "us-central1-b"));
+    DetectedPlatform mockServerlessPlatform = Mockito.mock(DetectedPlatform.class);
+    Mockito.when(mockServerlessPlatform.getSupportedPlatform()).thenReturn(platform);
+    Mockito.when(mockServerlessPlatform.getAttributes()).thenReturn(mockAttributes);
+    Mockito.when(mockServerlessPlatform.getProjectId()).thenReturn(DUMMY_PROJECT_ID);
+    return mockServerlessPlatform;
+  }
+
+  private static DetectedPlatform generateMockGaePlatform() {
+    Map<String, String> mockAttributes =
+        new HashMap<>(
+            ImmutableMap.of(
+                GAE_MODULE_NAME, "gae-app",
+                GAE_APP_VERSION, "v1",
+                GAE_INSTANCE_ID, "gae-instance-id",
+                GAE_CLOUD_REGION, "us-central1",
+                GAE_AVAILABILITY_ZONE, "us-central1-b"));
+    DetectedPlatform mockGAEPlatform = Mockito.mock(DetectedPlatform.class);
+    Mockito.when(mockGAEPlatform.getSupportedPlatform())
+        .thenReturn(GCPPlatformDetector.SupportedPlatform.GOOGLE_APP_ENGINE);
+    Mockito.when(mockGAEPlatform.getAttributes()).thenReturn(mockAttributes);
+    Mockito.when(mockGAEPlatform.getProjectId()).thenReturn(DUMMY_PROJECT_ID);
+    return mockGAEPlatform;
+  }
+
+  private static DetectedPlatform generateMockUnknownPlatform() {
+    Map<String, String> mockAttributes =
+        new HashMap<>(
+            ImmutableMap.of(
+                GCE_INSTANCE_ID, "instance-id",
+                GCE_CLOUD_REGION, "australia-southeast1"));
+
+    DetectedPlatform mockUnknownPlatform = Mockito.mock(DetectedPlatform.class);
+    Mockito.when(mockUnknownPlatform.getSupportedPlatform())
+        .thenReturn(GCPPlatformDetector.SupportedPlatform.UNKNOWN_PLATFORM);
+    Mockito.when(mockUnknownPlatform.getAttributes()).thenReturn(mockAttributes);
+    return mockUnknownPlatform;
+  }
+
+  @Test
+  public void testGceResourceAttributesMapping() {
+    GCPPlatformDetector mockDetector = Mockito.mock(GCPPlatformDetector.class);
+    DetectedPlatform mockPlatform = generateMockGcePlatform();
+    Mockito.when(mockDetector.detectPlatform()).thenReturn(mockPlatform);
+
+    Resource gotResource = new GCPResourceProvider(mockDetector).createResource(mockConfigProps);
+    Mockito.verify(mockPlatform, Mockito.times(1)).getProjectId();
+
+    assertEquals(
```

Comment:
[minor nitpick] maybe switch to using assertJ `assertThat` here as it's widely used elsewhere in the project, also it provides a convenient syntax for checking map entries.